File: python-leetcode/11.01.py
class Solution:
  def maxArea(self, height: List[int]) -> int:
    result = 0
    for i in range(len(height)-1):
      for j in range(i + 1, len(height)):
        curHeight = min(height[i], height[j])
        curWdith = j - i
        curArea = curWdith * curHeight
        result = max(result, curArea)
    return result

File: cpp-leetcode/11.01.cpp
#include <iostream>
#include <vector>
#include <utility>
using namespace std;

class Solution {
public:
  int maxArea(vector<int>& height) {
    int result = 0;
    for (int i = 0; i < height.size() - 1; i++) {
      for (int j = 0; j < height.size(); j++) {
        int curHeight = min(height[i], height[j]);
        int curWidth = j - i;
        int curArea = curHeight * curWidth;
        result = max(result, curArea);
      }
    }
    return result;
  }
};

int main() {
  vector<int> height = {1,8,6,2,5,4,8,3,7};
  Solution o;
  int result = o.maxArea(height);
  printf("result = %d\n", result);
  return 0;
}
